src/components/Router/StepComponentMap.js

```diff
@@ -1,58 +1,44 @@
-  const stepCount = formattedStepOptions.length
-  const mapSteps = (step, index) => {
-    const options = {step, componentOptions}
-    if (stepCount === index + 1) {
-      options.finalStep = true
-    }
-    return createComponent(options)
-  }
-  return shallowFlatten(formattedStepOptions.map(mapSteps))
-}
-
-const createDocumentComponents = (options, finalStep) => {
-  const finalOptions = finalStep ? {...options, nextStep: onFinalStep} : options
-  const double_sided_docs = ['driving_licence', 'national_identity_card']
-  if (double_sided_docs.indexOf(options.documentType) !== -1) {
-    return [<Select {...options}/>,
-            <FrontDocumentCapture {...options}/>,
-            <BackDocumentCapture {...finalOptions}/>]
-  }
-  return [<Select {...options}/>, <FrontDocumentCapture {...finalOptions}/>]
+export const createComponentList = (steps, documentType) => {
+  const mapSteps = (step) => createComponent(step, documentType)
+  return shallowFlatten(steps.map(mapSteps))
 }
 
-const createComponent = ({step, componentOptions, finalStep}) => {
-  const options = {...step.options, ...componentOptions}
-  if (finalStep && step.type !== 'document') options.nextStep = onFinalStep
+const createComponent = (step, documentType) => {
+  const wrapComponentB = wrapComponent(step)
   const stepMap = {
-    welcome: () => <Welcome {...options}/>,
-    face: () => <FaceCapture {...options}/>,
-    document: () => createDocumentComponents(options, finalStep),
-    complete: () => <Complete {...options} />
+    welcome: () => wrapComponentB(Welcome),
+    face: () => wrapComponentB(FaceCapture, `capture`),
+    document: () => createDocumentComponents(step, documentType),
+    complete: () => wrapComponentB(Complete)
   }
-  if (!(step.type in stepMap)) { console.error('No such step: ' + step.type) }
-  return stepMap[step.type]()
+  const {type} = step
+  if (!(type in stepMap)) { console.error('No such step: ' + type) }
+  return stepMap[type]()
 }
 
-const onFinalStep = () => {
-  events.emit('complete')
+const createDocumentComponents = (step, documentType) => {
+  const wrapComponentB = wrapComponent(step)
+  const double_sided_docs = ['driving_licence', 'national_identity_card']
+  const frontDocumentFlow = [
+    wrapComponentB(Select, "type_select"),
+    wrapComponentB(FrontDocumentCapture, "capture_front")]
+
+  if (double_sided_docs.indexOf(documentType) !== -1) {
+    return [
+      ...frontDocumentFlow,
+      wrapComponentB(BackDocumentCapture, "capture_back")]
+  }
+  return frontDocumentFlow
 }
 
-// {type} will not return an object with the property type, because {} are also
-// used to establish a multi line function
-const typeToStep = type => {return {type}}
-
-const isStep = val => typeof val === 'object'
-
-const formatStep = typeOrStep => isStep(typeOrStep) ?  typeOrStep : typeToStep(typeOrStep)
+const wrapComponent = ({options: stepOption, type: stepType}) =>
```

Comment:
`wrapComponentB` could definitely be renamed. Also what does wrap mean in this case? And what does B stand for?